extensions/Worker.Extensions.Http.AspNetCore.Analyzers/src/CodeFixForHttpResultAttributeExpected.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CodeFixForHttpResultAttribute)), Shared]
+    public sealed class CodeFixForHttpResultAttribute : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(DiagnosticDescriptors.MultipleOutputHttpTriggerWithoutHttpResultAttribute.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Diagnostic diagnostic = context.Diagnostics.First();
+            context.RegisterCodeFix(new AddHttpResultAttribute(context.Document, diagnostic), diagnostic);
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// CodeAction implementation which adds the HttpResultAttribute on the return type of a function using the multi-output bindings pattern.
+        /// </summary>
+        private sealed class AddHttpResultAttribute : CodeAction
+        {
+            private readonly Document _document;
+            private readonly Diagnostic _diagnostic;
+            private const string ExpectedAttributeName = "HttpResult";
+
+
+            internal AddHttpResultAttribute(Document document, Diagnostic diagnostic)
+            {
+                this._document = document;
+                this._diagnostic = diagnostic;
+            }
+
+            public override string Title => "Add HttpResultAttribute";
+
+            public override string EquivalenceKey => null;
+
+            /// <summary>
+            /// Returns an updated Document with HttpResultAttribute added to the relevant property.
+            /// </summary>
+            protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
```

Comment:
nit: a proper use of the summary comment would be to include the returns and param properties as such:

```csharp
/// <summary>
/// Asynchronously retrieves the modified <see cref="Document"/>, with the HttpResultAttribute added to the relevant property.
/// </summary>
/// <param name="cancellationToken">A token that can be used to propagate notifications that the operation should be canceled.</param>
/// <returns>An updated <see cref="Document"/> object.</returns>
```